exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java

```diff
@@ -0,0 +1,249 @@
+    NetworkEvent networkEvent = timedEvent.getEvent();
+    timeEventBuilder.setMessageEvent(
+        TimeEvent.MessageEvent.newBuilder()
+            .setId(networkEvent.getMessageId())
+            .setCompressedSizeBytes(networkEvent.getCompressedMessageSize())
+            .setUncompressedSizeBytes(networkEvent.getUncompressedMessageSize())
+            .setType(toMessageEventTypeProto(networkEvent))
+            .build());
+    return timeEventBuilder.build();
+  }
+
+  private static TimeEvent.MessageEvent.Type toMessageEventTypeProto(NetworkEvent networkEvent) {
+    if (networkEvent.getType() == Type.RECV) {
+      return MessageEvent.Type.RECEIVED;
+    } else {
+      return MessageEvent.Type.SENT;
+    }
+  }
+
+  // These are the attributes of the Span, where usually we may add more attributes like the agent.
+  private static Attributes toAttributesProto(
+      io.opencensus.trace.export.SpanData.Attributes attributes) {
+    Attributes.Builder attributesBuilder =
+        toAttributesBuilderProto(
+            attributes.getAttributeMap(), attributes.getDroppedAttributesCount());
+    attributesBuilder.putAttributeMap(AGENT_LABEL_KEY, AGENT_LABEL_VALUE);
+    return attributesBuilder.build();
+  }
+
+  private static Attributes.Builder toAttributesBuilderProto(
+      Map<String, io.opencensus.trace.AttributeValue> attributes, int droppedAttributesCount) {
+    Attributes.Builder attributesBuilder =
+        Attributes.newBuilder().setDroppedAttributesCount(droppedAttributesCount);
+    for (Map.Entry<String, io.opencensus.trace.AttributeValue> label : attributes.entrySet()) {
+      attributesBuilder.putAttributeMap(label.getKey(), toAttributeValueProto(label.getValue()));
+    }
+    return attributesBuilder;
+  }
+
+  private static Status toStatusProto(io.opencensus.trace.Status status) {
+    Status.Builder statusBuilder = Status.newBuilder().setCode(status.getCanonicalCode().value());
+    if (status.getDescription() != null) {
+      statusBuilder.setMessage(status.getDescription());
+    }
+    return statusBuilder.build();
+  }
+
+  private static TruncatableString toTruncatableStringProto(String string) {
+    return TruncatableString.newBuilder().setValue(string).setTruncatedByteCount(0).build();
+  }
+
+  private static com.google.protobuf.Timestamp toTimestampProto(Timestamp timestamp) {
+    return com.google.protobuf.Timestamp.newBuilder()
+        .setSeconds(timestamp.getSeconds())
+        .setNanos(timestamp.getNanos())
+        .build();
+  }
+
+  private static AttributeValue toAttributeValueProto(
+      io.opencensus.trace.AttributeValue attributeValue) {
+    final AttributeValue.Builder attributeValueBuilder = AttributeValue.newBuilder();
+    attributeValue.match(
+        new Function<String, Void>() {
+          @Override
+          public Void apply(String stringValue) {
+            attributeValueBuilder.setStringValue(toTruncatableStringProto(stringValue));
```

Comment:
Truncate to 256

Reply:
https://github.com/census-instrumentation/opencensus-java/issues/813